docs/content/references/sui-api/graphql/guides.mdx

```diff
@@ -0,0 +1,257 @@
+
+## Querying the service
+
+### cURL like querying
+
+The endpoint can also be queried using using REST calls by passing the query to the call:
+```shell
+# basic query
+curl -X POST https://graphql-beta.mainnet.sui.io \
+     --header "Content-Type: application/json" \
+     --data '{
+          "query": "query { epoch { referenceGasPrice } }"
+     }'
+# query with variables
+curl -X POST https://graphql-beta.mainnet.sui.io \
+     --header "Content-Type: application/json" \
+     --data '{
+  "query": "query ($epochID: Int!) { epoch(id: $epochID) { referenceGasPrice } }",
+  "variables": {
+    "epochID": 123
+  }
+}'
+```
+
+### Working with variables
+
+Variables are a powerful way for passing different ObjectIDs, Digests, Sui Addresses, and other kind of data for full flexibility. Below is an example for querying the reference gas price for epoch 100, by passing the epoch as a variable:
+
+```graphql
+query ($epochID: Int!) {
+  epoch(id: $epochID) {
+    referenceGasPrice
+  }
+}
+variables {
+   "epochID": 100
+}
+
+And the result is:
+
+```json
+{
+  "data": {
+    "epoch": {
+      "referenceGasPrice": "780"
+    }
+  }
+}
+```
+
+:::info
+When working with the online GraphQL IDE, paste the variables' values in the `Variables` pane on the bottom left of the window, without the `variables` keyword.
+:::
+
+### Multiple data in one query
+A less ideal situation that rose when using JSON RPC was when data needed was spread across multiple queries. The user needs to execute these queries one by one. With GraphQL, the user can specify which fields are needed, and get that information. For example, the following query retrieves the first 20 transaction blocks (and the sender's address, the gas object used to pay, the gas price, and buget) after a specific one, at epoch 97. In JSON RPC this would have required multiple steps.
+
+```graphql
+# Fetch the first 20 transactions after 231220100 for epoch 97
+{
+  epoch(id:97) {
+    transactionBlockConnection(first: 20, after:"231220100") {
+      pageInfo {
+        hasNextPage
+        endCursor
+      }
+      edges {
+        cursor
+        node {
+          digest
+          sender {
+            address
+          }
+          effects {
+            gasEffects {
+              gasObject {
+                address
+              }
+            }
+          }
+          gasInput {
+            gasPrice
+            gasBudget
+          }
+        }
+      }
+    }
+  }
+}
+  
+```
+
+
+## Migrating guides
+
+### From JSON RPC API
+
+The JSON RPC API provided several individual endpoints to get specific data (e.g., `get_totalTransactionBlocks`). While this can be achieved similarly with GraphQL, it is more powerful to determine the data needed in one query that involves multiple aspects (e.g., transactions, balance, coins), execute that query, and get the results. This guide showcases how to use GraphQL instead of JSON RPC on a few original endpoints.
```

Comment:
```suggestion
The JSON RPC API provided several individual endpoints to get specific data (for example, `get_totalTransactionBlocks`). While this can be achieved similarly with GraphQL, it is more powerful to determine the data you need in one query that involves multiple aspects (for example, transactions, balance, coins), execute that query, and get the results. This guide showcases how to use GraphQL instead of JSON RPC on a few original endpoints.
```